### PR TITLE
ref(performance): Add missing types to performance widgets

### PR DESCRIPTION
**Pull request by scttcper**
a lot of these were really similar GenericPerformanceWidgetProps components

part of https://github.com/getsentry/frontend-tsc/issues/79

### CODE REVIEW

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx [hunk 1]
```diff
@@ -741,60 +752,64 @@ export function LineChartListWidget(props: PerformanceWidgetProps) {
...
-              setSelectedIndex={setSelectListIndex}
-              items={getItems(provided)}
-            />
-          ),
-          height: 124,
-          noPadding: true,
-        },
-      ];
+  const Visualizations: GenericPerformanceWidgetProps<DataType>['Visualizations'] =
+    organization.features.includes('performance-new-widget-designs')
```
Comment (scttcper) [inline static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:756]: lots of the changes in this pr should be just the whitespace changes

File: static/app/views/performance/landing/widgets/widgets/mobileReleaseComparisonListWidget.tsx [hunk 2]
```diff
@@ -117,13 +122,15 @@ function MobileReleaseComparisonListWidget(props: PerformanceWidgetProps) {
...
-        }[props.chartSetting];
+        const sortField: string = (
+          {
+            [PerformanceWidgetSetting.SLOW_SCREENS_BY_TTID]: 'count()',
+            [PerformanceWidgetSetting.SLOW_SCREENS_BY_COLD_START]:
+              'count_starts(measurements.app_start_cold)',
+            [PerformanceWidgetSetting.SLOW_SCREENS_BY_WARM_START]:
+              'count_starts(measurements.app_start_warm)',
+          } as any
+        )[props.chartSetting];
```
Comment (scttcper) [inline static/app/views/performance/landing/widgets/widgets/mobileReleaseComparisonListWidget.tsx:133]: can't cast this to keyof itself

File: static/app/views/performance/landing/widgets/widgets/stackedAreaChartListWidget.tsx [hunk 3]
```diff
@@ -178,17 +187,17 @@ export function StackedAreaChartListWidget(props: PerformanceWidgetProps) {
...
 
-  const getAreaChart = provided => {
-    const durationUnit = getDurationUnit(provided.widgetData.chart.data);
+  const getAreaChart = (provided: ComponentData) => {
+    const durationUnit = getDurationUnit(provided.widgetData.chart.data ?? []);
     return (
       <StackedAreaChart
         {...provided.widgetData.chart}
-        {...provided}
+        {...(provided as any)}
```
Comment (scttcper) [inline static/app/views/performance/landing/widgets/widgets/stackedAreaChartListWidget.tsx:198]: the types here don't totally line up and it was previously `any`

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx [hunk 4]
```diff
@@ -630,11 +641,11 @@ export function LineChartListWidget(props: PerformanceWidgetProps) {
...
-          const description: string = listItem[SpanMetricsField.SPAN_DESCRIPTION];
-          const group: string = listItem[SpanMetricsField.SPAN_GROUP];
-          const projectID: number = listItem['project.id'];
-          const timeSpentPercentage: number = listItem[fieldString];
-          const totalTime: number = listItem[`sum(${SpanMetricsField.SPAN_SELF_TIME})`];
+          const description = listItem[SpanMetricsField.SPAN_DESCRIPTION] as string;
+          const group = listItem[SpanMetricsField.SPAN_GROUP] as string;
+          const projectID = listItem['project.id'] as number;
+          const timeSpentPercentage = listItem[fieldString] as number;
+          const totalTime = listItem[`sum(${SpanMetricsField.SPAN_SELF_TIME})`] as number;
```
Comment (gggritso) [inline static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:648]: Out of curiosity, why the change to `as` here?

**Review by gggritso — APPROVED** [review]
👍🏻

File: static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx (hunk 4 above)
Comment (scttcper) [reply static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:648]: before the type was
`const totalTime: number = any` which was fine. But now it's `const totalTime = string | number` so it has to be casted to number. I could've left the `const totalTime: number` i guess

Comment (gggritso) [reply static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx:648]: K, just curious!
